src/ducks/transactions/Transactions.jsx

```diff
@@ -192,28 +192,74 @@ const groupByDateAndSort = transactions => {
   return sortBy(toPairs(byDate), x => x[0]).reverse()
 }
 
+class ScrollSpy {
+  constructor(getScrollingElement) {
+    this.getScrollingElement = getScrollingElement
+    this.nodes = {}
+  }
+
+  addNode(id, node) {
+    this.nodes[id] = node
+  }
+
+  getTopMostVisibleNodeId() {
+    const scrollEl = this.getScrollingElement()
+    const topRoot =
+      scrollEl === window ? 0 : scrollEl.getBoundingClientRect().top
+    const offsets = sortBy(
+      Object.entries(this.nodes).map(([tId, node]) => [
+        tId,
+        node ? node.getBoundingClientRect().top : Infinity
+      ]),
+      x => x[1]
+    )
+    const topMost = find(offsets, o => {
+      const offset = o[1]
+      return offset - topRoot > 0
+    })
+    return topMost ? topMost[0] : null
+  }
+}
+
 class TransactionsD extends React.Component {
   state = {
     infiniteScrollTop: false
   }
 
   constructor(props) {
     super(props)
+    this.scrollSpy = new ScrollSpy(this.getScrollingElement)
   }
 
+  componentWillMount() {
+    this.updateTransactions(this.props.transactions)
   }
 
+  updateTransactions(transactions) {
+    this.transactionsById = keyBy(transactions, '_id')
+    this.transactions = transactions
   }
 
+  updateTopMostVisibleTransaction() {
+    const topMostTransactionId = this.scrollSpy.getTopMostVisibleNodeId()
+    const topMostTransaction = this.transactionsById[topMostTransactionId]
+    if (topMostTransaction) {
+      this.props.onChangeTopMostTransaction(topMostTransaction)
+    }
   }
 
   handleScroll = throttle(
     getScrollInfo => {
       this.props.onScroll(getScrollInfo)
+      this.updateTopMostVisibleTransaction()
     },
     100,
     { leading: false, trailing: true }
   )
+
+  handleRefRow = (transactionId, ref) => {
+    const node = ReactDOM.findDOMNode(ref) // eslint-disable-line
```

Comment:
`ref` is not supposed to be the `node` already ?

Reply:
a ref can be an instance of a React component.

> When the ref attribute is used on an HTML element, the ref created in the constructor with React.createRef() receives the underlying DOM element as its current property.
When the ref attribute is used on a custom class component, the ref object receives the mounted instance of the component as its current.

https://reactjs.org/docs/refs-and-the-dom.html